packages/@sanity/form-builder/src/inputs/Slug/SlugInput.js

```diff
@@ -55,14 +56,14 @@ export default withDocument(
       this._isMounted = false
     }
 
-    updateValue(value) {
-      if (!value) {
-        this.props.onChange(PatchEvent.from(unset()))
+    updateCurrent(current) {
+      const {onChange, type} = this.props
+      if (!current) {
+        onChange(PatchEvent.from(unset(['current'])))
```

Comment:
Won't this cause the object to remain? Isn't it generally a recommended pattern to unset the whole object when it is empty?

Reply:
The problem appears when editing slugs contained in an array and the user removes the last character in the input. Then the editing dialog will suddenly (and unexpectedly, IMO) be closed cause the item gets removed.
The pattern should rather be to check for emptiness when the dialog closes, but that is a concern for the array input I'd say.

Reply:
Oh, btw, I don't mean to say this is how it _should_ work, I'm rather describing how it works now. I'm all for a discussion about collapsing empty values, but there are quite a few considerations we need to do before supporting it.

Reply:
Interesting problem. When it is *not* inside of an array, you'll be left with a whole bunch of empty `{_type: "slug"}` objects, which isn't great either?

Reply:
Also, take the unicorn slider from the custom input widgets example. It generates an unset patch on empty values for the current path. Won't that also cause the same issue?

Reply:
This PR will fix a bunch of issues regardless of this conversation, so let's continue somewhere else.

Reply:
👍 agreed, let's talk tomorrow.